x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/RolloverAction.java

```diff
@@ -71,6 +72,11 @@ public RolloverAction(@Nullable ByteSizeValue maxSize, @Nullable ByteSizeValue m
         if (maxSize == null && maxPrimaryShardSize == null && maxAge == null && maxDocs == null) {
             throw new IllegalArgumentException("At least one rollover condition must be set.");
         }
+
+        if (maxDocs != null && maxDocs > IndexWriter.MAX_DOCS) {
+            throw new IllegalArgumentException("max_docs cannot exceed Lucene limit.");
+        }
```

Comment:
I believe we should convert this to a warning message and update `maxDocs` to `IndexWriter.MAX_DOCS` (treat it as if it's been configured to the max value) as otherwise existing policies that'd trigger this will not be able to be loaded in the cluster state (preventing nodes from starting).

The validation of the `PutLifecycle.Request` should be enhanced to add a validation error for this case (so we prevent new policies being created or updated to an invalid state) https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/action/PutLifecycleAction.java#L63

Reply:
Thanks for the comments, will be pushing the changes in the upcoming commits.